lib/index.js

```diff
@@ -120,46 +120,74 @@ Optimizely.prototype.page = function(page) {
  *
  */
 
 Optimizely.prototype.sendClassicDataToSegment = function(experimentState) {
   var experiment = experimentState.experiment;
   var variations = experimentState.variations;
-  var section = experimentState.section;
+  var sections = experimentState.sections;
   var variationNames = [];
+  var variationIds = [];
 
   // Makes concatenating easier later
   each(function(variation) {
     variationNames.push(variation.name);
+    variationIds.push(variation.id);
   }, variations);
 
   // Send data via `.track()`
   if (this.options.listen) {
     var props = {
       experimentId: experiment.id, // to be consistent with other ids
-      experimentName: experiment.name
+      experimentName: experiment.name,
+      variationName: variationNames.join(', '),
+      variationId: variationIds.join()
     };
 
-    // If this was a redirect experiment this value is made available
+    // If this was a redirect experiment and the effective referrer is different from document.referrer,
+    // this value is made available
     if (experiment.referrer) props.referrer = experiment.referrer;
 
-    // When only one variation running:
-    if (variations.length && variations.length < 2) {
-      props.variationName = variations[0].name;
-      props.variationId = variations[0].id;
-    }
-
     // When there is a multivariate experiment
-    if (variations.length > 1) {
-      props.variationName = variationNames.join(', '); // eg. 'Variation #1, Variation #2, Var 3709'
-      if (section) {
-        props.sectionName = section.name; // "grouping" of variations
-        props.variationId = section.variation_ids.join(); // eg '1,22,3709'
+    if (sections) {
+      // Since `sections` include all the possible sections on the page, we need to find the names of the sections
+      // if any of its variations were used. Experiments could display variations from multiple sections.
+      // So we have to do these terrible for loops as a workaround.
+      var sectionNames = [];
+      var sectionIds = [];
+      var sectionMap = {};
+      var activeSections = {};
+
+      // We need to reformat the data structure to reduce complexity from 4x to 2x
+      for (var sectionId in sections) {
+        if (!sections.hasOwnProperty(sectionId)) continue;
+        var section = sections[sectionId];
+        var sectionVariationIds = section.variation_ids;
+        for (var i = 0; i < sectionVariationIds.length; i++) {
+          sectionMap[sectionVariationIds[i]] = {
+            id: sectionId,
+            name: section.name
+          };
+        }
       }
+
+      // Experiments can display variations from multiple sections so we are checking which
+      // sections were actually used
+      for (var j = 0; j < variationIds.length; j++) {
+        var activeSection = sectionMap[variationIds[j]];
+        if (activeSection && !activeSections[activeSection.id]) {
```

Comment:
Finally, is it guaranteed that variationIds are unique? That is, it's not possible for section 1 to have a variation with id 1 and section 2 to also have a variation with id 1, right?